### DISCUSSION TITLE
Facets type slider alway visible without data

### DISCUSSION THREAD

**Discussion by mlafrance-absolunet** [algolia/instantsearch#5592] [General] [answered]
Hi Algolia team,

I have already open request [#551074](https://support.algolia.com/hc/en-us/requests/551074), and I got the answer to come here.

Current configs : 
- PHP 8.1 | algolia/algoliasearch-magento-2 3.9.1 | Magento 2.4.5-p2
- Magento admin > Instant Search Results Page > Facets >
  - Added product attributes with integer data value with type (Slider) searcheable

When i try search result with examples products without data with my custom attributes, this is always visible on filters with [0 - 0].

In Dashboard Algolia (indexes data), i validate this data is here only for good products. Example [this simple product](https://www.meublesrd.com/en/catalogsearch/result/?q=039339), have data for attribute `WIDTH (INCHES)` , else if i try search product without this data in same attribute, this facet always visible.

- Example 1 with [this screenshot](https://p17.zdusercontent.com/attachment/10807005/Mz75Qt7XJ8sj81W2rMIqYnI05) without custom code (navively)
- Exemple 2 live from [this link](https://www.meublesrd.com/en/catalogsearch/result/?q=nodata) with small code js/css by mixin added.

Thank for help if you can add fix on module.

Cordialement,
Matthew Lafrance (Valtech)

**Comment by Haroenv** [comment, marked as answer]
You can add a panel around the widget to hide it based on a condition:

```js
instantsearch.widgets.panel({
  hidden(options) {
    return options.range.min === 0 && options.range.max === 0;
  },
})(instantsearch.widgets.rangeSlider)({
  container: '#price',
  attribute: 'price',
});
```

https://codesandbox.io/s/happy-fermat-ezupux?file=/src/app.js

**Reply by mlafrance-absolunet** [reply]
Thank you @Haroenv, it works well. Have a nice day.
